ui/src/app/applications/components/application-status-panel/application-status-panel.tsx

```diff
@@ -58,60 +58,124 @@ const sectionHeader = (info: SectionInfo, onClick?: () => any) => {
     );
 };
 
-const hasRollingSyncEnabled = (application: models.Application): boolean => {
-    return application.metadata.ownerReferences?.some(ref => ref.kind === 'ApplicationSet') || false;
+const getApplicationSetOwnerRef = (application: models.Application) => {
+    return application.metadata.ownerReferences?.find(ref => ref.kind === 'ApplicationSet');
 };
 
-const ProgressiveSyncStatus = ({application}: {application: models.Application}) => {
-    if (!hasRollingSyncEnabled(application)) {
-        return null;
-    }
+const extractErrorText = (error: any): string => {
+    const errorChildren = error.props?.children || [];
+    return Array.isArray(errorChildren) ? errorChildren.map(child => (typeof child === 'string' ? child : '')).join('') : String(errorChildren);
+};
 
-    const appSetRef = application.metadata.ownerReferences.find(ref => ref.kind === 'ApplicationSet');
+const isPermissionError = (error: any): boolean => {
+    const errorText = extractErrorText(error);
+    return errorText.includes('Failed to load data');
```

Comment:
Relying on error text is fragile, could you look into the error object when permissionError occurs to see if something like errorCode or statusCode(403) or error type exists and use that instead?